src/components/Amalgamation/ResultingBusinessName.vue

```diff
@@ -0,0 +1,193 @@
+            @cancel="resetName()"
+            @update:companyName="onUpdateCompanyName($event)"
+            @update:formType="formType = $event"
+            @update:nameRequest="onUpdateNameRequest($event)"
+          />
+        </v-col>
+      </v-row>
+    </div>
+
+    <!-- Display Mode -->
+    <template v-else>
+      <NameRequestInfo />
+      <NameTranslations class="mt-n8" />
+
+      <v-btn
+        text
+        color="primary"
+        class="btn-undo"
+        @click="resetName()"
+      >
+        <v-icon small>
+          mdi-undo
+        </v-icon>
+        <span>Undo</span>
+      </v-btn>
+    </template>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { Component, Mixins } from 'vue-property-decorator'
+import { Getter, Action } from 'pinia-class'
+import { useStore } from '@/store/store'
+import { NameRequestMixin } from '@/mixins'
+import { EmptyNameRequest, NameRequestIF } from '@/interfaces/'
+import { LegalServices } from '@/services/'
+import { CorrectNameOptions, NrRequestActionCodes } from '@bcrs-shared-components/enums/'
+import { CorpTypeCd } from '@bcrs-shared-components/corp-type-module'
+import { CorrectName } from '@bcrs-shared-components/correct-name/'
+import NameRequestInfo from '@/components/common/NameRequestInfo.vue'
+import NameTranslations from '@/components/common/NameTranslations.vue'
+
+@Component({
+  components: {
+    CorrectName,
+    NameRequestInfo,
+    NameTranslations
+  }
+})
+export default class ResultingBusinessName extends Mixins(NameRequestMixin) {
+  @Getter(useStore) getBusinessId!: string
+  @Getter(useStore) getBusinessLegalName!: string
+  @Getter(useStore) getCorrectNameOption!: CorrectNameOptions
+  @Getter(useStore) getEntityType!: CorpTypeCd
+  @Getter(useStore) getNameRequest!: NameRequestIF
+  @Getter(useStore) getNameRequestApprovedName!: string
+  @Getter(useStore) getNameRequestNumber!: string
+  @Getter(useStore) getShowErrors!: boolean
+
+  @Action(useStore) setCorrectNameOption!: (x: CorrectNameOptions) => void
+  @Action(useStore) setNameRequest!: (x: NameRequestIF) => void
+  @Action(useStore) setNameRequestApprovedName!: (x: string) => void
+
+  // Local properties
+  businessNameOption = null as string
+  // businessNameOption = this.getNameRequestNumber ? 'named' : 'numbered'
+  formType = null as CorrectNameOptions
+
+  readonly correctionNameChoices = [
+    CorrectNameOptions.CORRECT_AML_ADOPT,
+    CorrectNameOptions.CORRECT_NEW_NR,
+    CorrectNameOptions.CORRECT_AML_NUMBERED
+  ]
+
+  /** The company name. */
+  get companyName (): string {
+    return (this.getNameRequestApprovedName || this.getBusinessLegalName)
+  }
+  /** This section's validity state (when prompted by app). */
+  get invalidSection (): boolean {
+    return (this.getShowErrors && !this.getCorrectNameOption)
+  }
+
+  /** Called when component is created. */
+  created (): void {
+    // this.businessNameOption = this.getNameRequestNumber ? 'named' : 'numbered'
```

Comment:
Eventually, we're going to have to comment this line out right Sev?

Reply:
Sort of. There will be a new filing variable to save/restore which option was selected (null initially in some cases). I'm not there yet :)

Reply:
Oops, sorry. All right Sev!